comm/src/main/scala/coop/rchain/comm/protocol.scala

```diff
@@ -48,20 +52,21 @@ class ProtocolNode(id: NodeIdentifier, endpoint: Endpoint, handler: ProtocolHand
     _seq
   }
 
-  override def ping: Try[Duration] =
-    ProtocolMessage.ping(handler.local) match {
-      case Some(ping) =>
-        handler.roundTrip(PingMessage(ping, System.currentTimeMillis), this) match {
-          case Success(pong) =>
-            ping.header match {
-              case Some(incoming) =>
-                Success(Duration(pong.timestamp - incoming.timestamp, MILLISECONDS))
-              case _ => Failure(new Exception("ping failed"))
-            }
-          case Failure(ex) => Failure(ex)
+  override def ping: Try[Duration] = {
+    val req = PingMessage(ProtocolMessage.ping(handler.local), System.currentTimeMillis)
+    handler.roundTrip(req, this) match {
+      case Right(resp) =>
+        req.header match {
+          case Some(incoming) =>
+            Success(Duration(resp.timestamp - incoming.timestamp, MILLISECONDS))
+          case _ => Failure(new Exception("ping failed"))
         }
-      case None => Failure(new Exception("ping failed"))
+      case Left(ex) => ex match {
+        case ProtocolException(exc) => Failure(exc)
+        case exc => Failure(new Exception(exc.toString))
```

Comment:
I'd prefer to use `Exception` as constructor argument, so we could have a full stack trace.